app/code/Magento/InventoryCatalog/Model/GetProductIdsBySkus.php

```diff
@@ -33,6 +34,15 @@ public function __construct(
      */
     public function execute(array $skus): array
     {
-        return $this->productResource->getProductsIdsBySkus($skus);
+        $idsBySkus = $this->productResource->getProductsIdsBySkus($skus);
```

Comment:
why is the exception not thrown out of `getProductIdsBySkus` method? 

We have `NoSuchEntityException` the exception thrown from most repositories on Get operations 
```
     * @throws \Magento\Framework\Exception\NoSuchEntityException
     */
    public function get($sku, $editMode = false, $storeId = null, $forceReload = false);
```

here is example for Product Repository.

Why did you decide to make this responsibility of Business Logic, but not service?